python/EFS_ENCRYPTED_CHECK/EFS_ENCRYPTED_CHECK.py

```diff
@@ -0,0 +1,450 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+    EFS_ENCRYPTED_CHECK
+Description:
+    Check whether Amazon EFS Filesytems are configured to encrypt the file data using AWS Key Management Service (AWS KMS).
+
+Trigger:
+    Periodic
+
+Reports on:
+    AWS::EFS::FileSystem
+
+Rule Parameters:
+    KmsKeyId
+     Optional
+     ARN of the KMS key that is used to encrypt the EFS filesystem
+
+Scenarios:
+  Scenario 1:
+  Given: No EFS filesystem is present
+   Then: Return NOT_APPLICABLE
+
+  Scenario 2:
+  Given: At least one EFS filesystem is present
+    And: The "Encrypted" key is set to False on DescribeFileSystems
+   Then: Return NON_COMPLIANT on this EFS Filesystem
+
+  Scenario 3:
+  Given: At least one EFS filesystem is present
+    And: The "Encrypted" key is set to True on DescribeFileSystems
+    And: KmsKeyId parameter is not configured
+   Then: Return COMPLIANT on this EFS Filesystem
+
+  Scenario 4:
+  Given: At least one EFS filesystem is present
+    And: The "Encrypted" key is set to True on DescribeFileSystems
+    And: KmsKeyId parameter is configured
+    And: KmsKeyId key on DescribeFileSystems is not matching KmsKeyId parameter.
+   Then: Return NON_COMPLIANT on this EFS Filesystem
+
+  Scenario 5:
+  Given: At least one EFS filesystem is present
+    And: The "Encrypted" key is set to True on DescribeFileSystems
+    And: KmsKeyId parameter is configured
+    And: KmsKeyId key on DescribeFileSystems is matching KmsKeyId parameter
+   Then: Return COMPLIANT on this EFS Filesystem
+"""
+
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::EFS::FileSystem'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    efs_client = get_client('efs', event)
+
+    # get all the file systems
+    all_file_systems = get_all_file_systems(efs_client)
+
+    evaluations = []
+
+    # check whether atlease one file system exist
+    if not all_file_systems:
+        return None
+
+    for each_efs in all_file_systems:
+        # check if file system is encrypted
+        if not each_efs['Encrypted']:
+            evaluations.append(build_evaluation(each_efs['FileSystemId'], 'NON_COMPLIANT', event, annotation='EFS is not encryprted'))

```

Comment:
Typo in the annotation. Please call the services by the full name. Add a final dot.